lib/signed-xml.js

```diff
@@ -399,19 +388,33 @@ SignedXml.prototype.getCanonReferenceXml = function(doc, ref, node) {
   return this.getCanonXml(ref.transforms, node, c14nOptions)
 }
 
-SignedXml.prototype.validateSignatureValue = function(doc) {
+SignedXml.prototype.validateSignatureValue = async function(doc) {
   var signedInfoCanon = this.getCanonSignedInfoXml(doc)
   var signer = this.findSignatureAlgorithm(this.signatureAlgorithm)
-  var res = signer.verifySignature(signedInfoCanon, this.signingKey, this.signatureValue)
-  if (!res) this.validationErrors.push("invalid signature: the signature value " +
-                                        this.signatureValue + " is incorrect")
-  return res
+  return new Promise((resolve) => {
+    signer.verifySignature(signedInfoCanon, this.signingKey, this.signatureValue, (err, res) => {
+      if (err || !res) {
+        this.validationErrors.push("invalid signature: the signature value " +
+            this.signatureValue + " is incorrect")
```

Comment:
Needs to be `this.signatureValue.toString('base64')` now we're using a buffer instead of string.